angular-intro-project/src/app/auth.service.ts

```diff
@@ -1,38 +1,43 @@
 import { Injectable } from '@angular/core';
 import { Subject } from 'rxjs';
+import { HttpClient } from '@angular/common/http';
+import { Router } from '@angular/router';
 
 @Injectable({
   providedIn: 'root'
 })
 export class AuthService {
-  currentEmail: string = '';
+  token: string = '';
   nameChange: Subject<string> = new Subject<string>();
-  constructor() { }
+  constructor(private httpClient: HttpClient, private router: Router) { }
 
-  login(email: string, pass: string): void {
-    const user = {
-      pass,
-      token: '123abc'
-    };
-    const serialUser = JSON.stringify(user);
-    localStorage.setItem(email, serialUser);
-    this.currentEmail = email;
-    this.nameChange.next(this.currentEmail);
-    console.log('Logged in');
+  login(login: string, password: string): void {
+    const body = { login, password };
+    this.httpClient.post('http://localhost:3004/auth/login', body)
```

Comment:
I would move `http://localhost:3004` into `environment` property in environment.ts file. In simple case it can be moved in some global constants, so we shouldn't use hard coded values.